From 59c70d00040ca06f9da34b78b38cf03da5cd268c Mon Sep 17 00:00:00 2001
From: Adria Lopez <adria.lopez.ortega@gmail.com>
Date: Fri, 26 Feb 2016 17:19:37 +0100
Subject: [PATCH 1/3] Cleanup class

I've cleaned some unused references and sorted the methods to place all
public methods before the private ones
---
 .../TimeoutPersister/MongoTimeoutPersister.cs      | 70 +++++++++++++---------
 1 file changed, 41 insertions(+), 29 deletions(-)

diff --git a/src/NServiceBus.MongoDB/TimeoutPersister/MongoTimeoutPersister.cs b/src/NServiceBus.MongoDB/TimeoutPersister/MongoTimeoutPersister.cs
index 3dc50f2..a1d7e09 100644
--- a/src/NServiceBus.MongoDB/TimeoutPersister/MongoTimeoutPersister.cs
+++ b/src/NServiceBus.MongoDB/TimeoutPersister/MongoTimeoutPersister.cs
@@ -33,8 +33,6 @@ namespace NServiceBus.MongoDB.TimeoutPersister
     using System.Diagnostics.Contracts;
     using System.Linq;
 
-    using global::MongoDB.Bson.Serialization.Attributes;
-    using global::MongoDB.Bson.Serialization.Options;
     using global::MongoDB.Driver;
     using global::MongoDB.Driver.Builders;
     using global::MongoDB.Driver.Linq;
@@ -196,6 +194,47 @@ public void RemoveTimeoutBy(Guid sagaId)
             }
         }
 
+        /// <summary>
+        /// Reads timeout data.
+        /// </summary>
+        /// <param name="timeoutId">The timeout id to read.</param>
+        /// <returns>
+        /// <see cref="T:NServiceBus.Timeout.Core.TimeoutData"/> of the timeout if it was found. <c>null</c> otherwise.
+        /// </returns>
+        public Timeout.Core.TimeoutData Peek(string timeoutId)
+        {
+            var collection = this.mongoDatabase.GetCollection<TimeoutData>(TimeoutDataName);
+            var data = collection.AsQueryable().SingleOrDefault(e => e.Id == timeoutId);
+            if (data != null)
+            {
+                return new Timeout.Core.TimeoutData()
+                {
+                    Id = data.Id,
+                    Destination = data.Destination,
+                    SagaId = data.SagaId,
+                    State = data.State,
+                    Time = data.Time,
+                    Headers = data.Headers,
+                    OwningTimeoutManager = data.OwningTimeoutManager
+                };
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// The IPersistTimeoutsV2 implementation of the TryRemove method
+        /// </summary>
+        /// <param name="timeoutId">The timeout id to remove.</param>
+        /// <returns>
+        /// <c>true</c> it the timeout was successfully removed.
+        /// </returns>
+        public bool TryRemove(string timeoutId)
+        {
+            Timeout.Core.TimeoutData timeoutData;
+            return this.TryRemove(timeoutId, out timeoutData);
+        }
+
         private void EnsureTimeoutIndexes()
         {
             var collection = this.mongoDatabase.GetCollection<TimeoutData>(TimeoutDataName);
@@ -219,32 +258,5 @@ private void ObjectInvariants()
         {
             Contract.Invariant(this.mongoDatabase != null);
         }
-
-        public Timeout.Core.TimeoutData Peek(string timeoutId)
-        {
-            var collection = this.mongoDatabase.GetCollection<TimeoutData>(TimeoutDataName);
-            var data = collection.AsQueryable().SingleOrDefault(e => e.Id == timeoutId);
-            if (data != null)
-            {
-                return new Timeout.Core.TimeoutData()
-                {
-                    Id = data.Id,
-                    Destination = data.Destination,
-                    SagaId = data.SagaId,
-                    State = data.State,
-                    Time = data.Time,
-                    Headers = data.Headers,
-                    OwningTimeoutManager = data.OwningTimeoutManager
-                };
-            }
-
-            return null;
-        }
-
-        public bool TryRemove(string timeoutId)
-        {
-            Timeout.Core.TimeoutData timeoutData;
-            return this.TryRemove(timeoutId, out timeoutData);
-        }
     }
 }
-- 
2.8.1.windows.1

